contracts/Quest.sol

```diff
@@ -250,4 +274,31 @@ contract Quest is ReentrancyGuardUpgradeable, PausableUpgradeable, Ownable, IQue
         uint erc20Balance = erc20Address_.balanceOf(address(this));
         if (erc20Balance > 0) erc20Address_.safeTransfer(msg.sender, erc20Balance);
     }
+
+    function _createLockupLinearStream(address recepient_, uint totalAmount_) internal {
+        // Approve the Sablier contract to spend reward tokens
+        rewardToken.safeApprove(address(sablierV2LockupLinearContract), totalAmount_);
+
+        LockupLinear.CreateWithDurations memory params;
+
+        params.sender = msg.sender; // The sender will be able to cancel the stream, this is the QuestFactory contract
+        params.recipient = recepient_; // The recipient of the streamed assets
+        params.totalAmount = uint128(totalAmount_); // Total amount is the amount inclusive of all fees
+        params.asset = IERC20(rewardToken); // The streaming asset
```

Comment:
If we're going to need to wrap this in an interface for every stream creation we might see some gas savings pushing this to an interface in the initialize call and saving that value. Not a cut and dry trade-off but something to consider.